src/compiler/scala/tools/nsc/settings/Warnings.scala

```diff
@@ -35,16 +35,35 @@ trait Warnings {
     "Configure reporting of compiler warnings; use `help` for details.",
     helpText = Some(
       s"""Configure compiler warnings.
-         |Syntax: -Wconf:<filter>&...&<filter>:<action>,<filter>:<action>,...
+         |Syntax: -Wconf:<filters>:<action>,<filters>:<action>,...
+         |where <filters> is one or more <filter> combined with &, i.e., <filter>&...&<filter>
+         |
+         |Note: Run with `-Wconf:any:warning-verbose` to print warnings with their the category,
+         |site, origin (for deprecations) and since version (for deprecations).
          |
          |<filter>
-         |  - any message: any
-         |  - message categories: cat=deprecation, cat=lint, cat=lint-infer-any
-         |  - message content (prefix / suffix are skipped): msg=regex
-         |  - site where the warning is triggered (regex must cover full name): site=my.package.*
-         |  - source file name (regex must cover full relative path): src=src_managed/*
-         |  - origin of deprecation (regex must cover full name): origin=external.package.*
-         |  - since of deprecation (operators: <, =, >, versions: n, n.m, n.m.p): since<1.24
+         |  - Any message: any
+         |
+         |  - Message categories: cat=deprecation, cat=lint, cat=lint-infer-any
+         |    The full list of warning categories is shown at the end of this help text.
+         |
+         |  - Message content: msg=regex
+         |    The regex must only match a part of the message, not necessarily the full message.
+         |
+         |  - Site where the warning is triggered: site=my.package.*
+         |    The regex must match the full name of the warning position.
+         |
+         |  - Source file name: src=src_managed/.*
+         |    The regex must match the canonical path, starting at any path segment.
```

Comment:
The documentation should definitely mention that file paths will always have file name separators normalized to Unix-style (`/`).